### ISSUE TITLE
Example code doesn't/no longer works

### ISSUE THREAD

**Issue by kwyntes** [maciejhirsz/logos#287]
Copy/pasted the example code from the README, following errors occur:
```
   Compiling logostest v0.1.0 (C:\Users\qjnri\source\rust\logostest)
error: Invalid nested attribute
  --> src\main.rs:31:9
   |
31 | #[logos(skip r"[ \t\n\f]+")] // Ignore this regex pattern between tokens       
   |         ^^^^

error: missing #[error] token variant.
  --> src\main.rs:30:10
   |
30 | #[derive(Logos, Debug, PartialEq)]
   |          ^^^^^
   |
   = note: this error originates in the derive macro `Logos` (in Nightly builds, run
 with -Z macro-backtrace for more info)                                             
error[E0046]: not all trait items implemented, missing: `ERROR`
  --> src\main.rs:30:10
   |
30 | #[derive(Logos, Debug, PartialEq)]
   |          ^^^^^ missing `ERROR` in implementation
   |
   = note: this error originates in the derive macro `Logos` (in Nightly builds, run
 with -Z macro-backtrace for more info)                                                = help: implement the missing item: `const ERROR: Self = value;`

error[E0308]: mismatched types
  --> src\main.rs:48:5
   |
48 |     assert_eq!(lex.next(), Some(Ok(Token::Text)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Token`, found 
enum `Result`
   |
   = note: expected enum `Option<Token>`
              found enum `Option<Result<Token, _>>`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run wi
th -Z macro-backtrace for more info)                                                
error[E0308]: mismatched types
  --> src\main.rs:52:5
   |
52 |     assert_eq!(lex.next(), Some(Ok(Token::Text)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Token`, found 
enum `Result`
   |
   = note: expected enum `Option<Token>`
              found enum `Option<Result<Token, _>>`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run wi
th -Z macro-backtrace for more info)                                                
error[E0308]: mismatched types
  --> src\main.rs:56:5
   |
56 |     assert_eq!(lex.next(), Some(Ok(Token::Fast)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Token`, found 
enum `Result`
   |
   = note: expected enum `Option<Token>`
              found enum `Option<Result<Token, _>>`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run wi
th -Z macro-backtrace for more info)                                                
error[E0308]: mismatched types
  --> src\main.rs:60:5
   |
60 |     assert_eq!(lex.next(), Some(Ok(Token::Text)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Token`, found 
enum `Result`
   |
   = note: expected enum `Option<Token>`
              found enum `Option<Result<Token, _>>`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run wi
th -Z macro-backtrace for more info)                                                
error[E0308]: mismatched types
  --> src\main.rs:64:5
   |
64 |     assert_eq!(lex.next(), Some(Ok(Token::Period)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `Token`, foun
d enum `Result`                                                                        |
   = note: expected enum `Option<Token>`
              found enum `Option<Result<Token, _>>`
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run wi
th -Z macro-backtrace for more info)                                                
Some errors have detailed explanations: E0046, E0308.
For more information about an error, try `rustc --explain E0046`.
```
Not sure what's happening.

**Comment by kwyntes** [comment]
I figured out one of the issues is that this was missing:
```rust
// ...
enum Token {
  // ...
  #[error]
  ERROR,
}
// ...
```

**Comment by maciejhirsz** [comment]
The README on GitHub is using some yet unpublished breaking changes. I should roll out a release soonTM.

Look at the [README on crates](https://crates.io/crates/logos) for up-to-date (down-to-date?) examples.

**Comment by maciejhirsz** [comment]
[Released 0.13](https://github.com/maciejhirsz/logos/releases/tag/v0.13)
